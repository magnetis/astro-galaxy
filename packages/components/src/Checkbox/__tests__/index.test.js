import React from 'react';
import { Label } from '@magnetis/astro-galaxy-components';
import { CheckboxWrapper, Checkbox, CheckboxShape } from '../index';

describe('Checkbox', () => {
  it('should render checkbox', () => {
    const json = rendererCreateWithTheme(
      <CheckboxWrapper>
        <Checkbox id="checkbox1" />
        <CheckboxShape />
        <Label htmlFor="checkbox1">Checkbox</Label>
      </CheckboxWrapper>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render a checked checkbox', () => {
    const json = rendererCreateWithTheme(
      <CheckboxWrapper>
        <Checkbox defaultChecked id="checkbox2" />
        <CheckboxShape />
        <Label htmlFor="checkbox2">Checked</Label>
      </CheckboxWrapper>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render a indeterminate checkbox', () => {
    const json = rendererCreateWithTheme(
      <CheckboxWrapper>
        <Checkbox indeterminate id="checkbox3" />
        <CheckboxShape />
        <Label htmlFor="checkbox3">Indeterminate</Label>
      </CheckboxWrapper>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render a disabled checkbox', () => {
    const json = rendererCreateWithTheme(
      <CheckboxWrapper>
        <Checkbox disabled id="checkbox4" />
        <CheckboxShape />
        <Label htmlFor="checkbox4">Disabled checkbox</Label>
      </CheckboxWrapper>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render a indeterminate disabled checkbox', () => {
    const json = rendererCreateWithTheme(
      <CheckboxWrapper>
        <Checkbox disabled indeterminate id="checkbox4" />
        <CheckboxShape />
        <Label htmlFor="checkbox4">Indeterminate disabled checkbox</Label>
      </CheckboxWrapper>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render a checked disabled checkbox', () => {
    const json = rendererCreateWithTheme(
      <CheckboxWrapper>
        <Checkbox defaultChecked disabled id="checkbox5" />
        <CheckboxShape />
        <Label htmlFor="checkbox5">Checked</Label>
      </CheckboxWrapper>
    ).toJSON();

    expect(json).toMatchSnapshot();
  });
});
