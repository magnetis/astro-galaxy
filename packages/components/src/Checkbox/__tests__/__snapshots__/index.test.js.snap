// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render a checked checkbox 1`] = `
.c3 {
  cursor: pointer;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  padding-left: 8px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input[type='checkbox']:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1 {
  position: absolute;
  z-index: 10;
  width: 16px;
  height: 16px;
  cursor: pointer;
  opacity: 0;
}

.c1:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1:focus ~ span::before {
  box-shadow: 0 0 0 3px rgba(21,156,228,0.4);
}

.c1:checked ~ span::before {
  background-color: #159ce4;
  border-color: #159ce4;
}

.c1:disabled ~ span::before {
  border-color: #e9eef2;
}

.c1:checked:disabled ~ span::before {
  background-color: #e9eef2;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:hover:not(:disabled) ~ span::before {
  border-color: #159ce4;
}

.c2::before {
  display: inline-block;
  width: 16px;
  height: 16px;
  content: '';
  border: 2px solid #597183;
  border-radius: 2px;
  position: relative;
  top: 2px;
}

.c2::after {
  position: absolute;
  top: 7px;
  left: 3px;
  width: 10px;
  height: 6px;
  color: #ffffff;
  content: '';
  border-bottom: 2px solid;
  border-left: 2px solid;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

@media (prefers-reduced-motion:reduce) {

}

<div
  className="c0"
>
  <input
    className="c1"
    defaultChecked={true}
    id="checkbox2"
    type="checkbox"
  />
  <span
    className="c2"
  />
  <label
    className="c3"
    htmlFor="checkbox2"
  >
    Checked
  </label>
</div>
`;

exports[`Checkbox should render a checked disabled checkbox 1`] = `
.c3 {
  cursor: pointer;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  padding-left: 8px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input[type='checkbox']:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1 {
  position: absolute;
  z-index: 10;
  width: 16px;
  height: 16px;
  cursor: pointer;
  opacity: 0;
}

.c1:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1:focus ~ span::before {
  box-shadow: 0 0 0 3px rgba(21,156,228,0.4);
}

.c1:checked ~ span::before {
  background-color: #159ce4;
  border-color: #159ce4;
}

.c1:disabled ~ span::before {
  border-color: #e9eef2;
}

.c1:checked:disabled ~ span::before {
  background-color: #e9eef2;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:hover:not(:disabled) ~ span::before {
  border-color: #159ce4;
}

.c2::before {
  display: inline-block;
  width: 16px;
  height: 16px;
  content: '';
  border: 2px solid #597183;
  border-radius: 2px;
  position: relative;
  top: 2px;
}

.c2::after {
  position: absolute;
  top: 7px;
  left: 3px;
  width: 10px;
  height: 6px;
  color: #ffffff;
  content: '';
  border-bottom: 2px solid;
  border-left: 2px solid;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

@media (prefers-reduced-motion:reduce) {

}

<div
  className="c0"
>
  <input
    className="c1"
    defaultChecked={true}
    disabled={true}
    id="checkbox5"
    type="checkbox"
  />
  <span
    className="c2"
  />
  <label
    className="c3"
    htmlFor="checkbox5"
  >
    Checked
  </label>
</div>
`;

exports[`Checkbox should render a disabled checkbox 1`] = `
.c3 {
  cursor: pointer;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  padding-left: 8px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input[type='checkbox']:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1 {
  position: absolute;
  z-index: 10;
  width: 16px;
  height: 16px;
  cursor: pointer;
  opacity: 0;
}

.c1:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1:focus ~ span::before {
  box-shadow: 0 0 0 3px rgba(21,156,228,0.4);
}

.c1:checked ~ span::before {
  background-color: #159ce4;
  border-color: #159ce4;
}

.c1:disabled ~ span::before {
  border-color: #e9eef2;
}

.c1:checked:disabled ~ span::before {
  background-color: #e9eef2;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:hover:not(:disabled) ~ span::before {
  border-color: #159ce4;
}

.c2::before {
  display: inline-block;
  width: 16px;
  height: 16px;
  content: '';
  border: 2px solid #597183;
  border-radius: 2px;
  position: relative;
  top: 2px;
}

.c2::after {
  position: absolute;
  top: 7px;
  left: 3px;
  width: 10px;
  height: 6px;
  color: #ffffff;
  content: '';
  border-bottom: 2px solid;
  border-left: 2px solid;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

@media (prefers-reduced-motion:reduce) {

}

<div
  className="c0"
>
  <input
    className="c1"
    disabled={true}
    id="checkbox4"
    type="checkbox"
  />
  <span
    className="c2"
  />
  <label
    className="c3"
    htmlFor="checkbox4"
  >
    Disabled checkbox
  </label>
</div>
`;

exports[`Checkbox should render a indeterminate disabled checkbox 1`] = `
.c3 {
  cursor: pointer;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  padding-left: 8px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input[type='checkbox']:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1 {
  position: absolute;
  z-index: 10;
  width: 16px;
  height: 16px;
  cursor: pointer;
  opacity: 0;
}

.c1:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1:focus ~ span::before {
  box-shadow: 0 0 0 3px rgba(21,156,228,0.4);
}

.c1:checked ~ span::before {
  background-color: #159ce4;
  border-color: #159ce4;
}

.c1:disabled ~ span::before {
  border-color: #e9eef2;
}

.c1:checked:disabled ~ span::before {
  background-color: #e9eef2;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:hover:not(:disabled) ~ span::before {
  border-color: #159ce4;
}

.c1 ~ span::after {
  top: 11px;
  left: 4px;
  width: 8px;
  border-left: 0;
  -webkit-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
}

.c1:disabled ~ span::before {
  background-color: #e9eef2;
}

.c2::before {
  display: inline-block;
  width: 16px;
  height: 16px;
  content: '';
  border: 2px solid #597183;
  border-radius: 2px;
  position: relative;
  top: 2px;
}

.c2::after {
  position: absolute;
  top: 7px;
  left: 3px;
  width: 10px;
  height: 6px;
  color: #ffffff;
  content: '';
  border-bottom: 2px solid;
  border-left: 2px solid;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

@media (prefers-reduced-motion:reduce) {

}

<div
  className="c0"
>
  <input
    className="c1"
    disabled={true}
    id="checkbox4"
    type="checkbox"
  />
  <span
    className="c2"
  />
  <label
    className="c3"
    htmlFor="checkbox4"
  >
    Indeterminate disabled checkbox
  </label>
</div>
`;

exports[`Checkbox should render an indeterminate checkbox 1`] = `
.c3 {
  cursor: pointer;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  padding-left: 8px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input[type='checkbox']:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1 {
  position: absolute;
  z-index: 10;
  width: 16px;
  height: 16px;
  cursor: pointer;
  opacity: 0;
}

.c1:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1:focus ~ span::before {
  box-shadow: 0 0 0 3px rgba(21,156,228,0.4);
}

.c1:checked ~ span::before {
  background-color: #159ce4;
  border-color: #159ce4;
}

.c1:disabled ~ span::before {
  border-color: #e9eef2;
}

.c1:checked:disabled ~ span::before {
  background-color: #e9eef2;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:hover:not(:disabled) ~ span::before {
  border-color: #159ce4;
}

.c1 ~ span::after {
  top: 11px;
  left: 4px;
  width: 8px;
  border-left: 0;
  -webkit-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
}

.c1:disabled ~ span::before {
  background-color: #e9eef2;
}

.c2::before {
  display: inline-block;
  width: 16px;
  height: 16px;
  content: '';
  border: 2px solid #597183;
  border-radius: 2px;
  position: relative;
  top: 2px;
}

.c2::after {
  position: absolute;
  top: 7px;
  left: 3px;
  width: 10px;
  height: 6px;
  color: #ffffff;
  content: '';
  border-bottom: 2px solid;
  border-left: 2px solid;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

@media (prefers-reduced-motion:reduce) {

}

<div
  className="c0"
>
  <input
    className="c1"
    id="checkbox3"
    type="checkbox"
  />
  <span
    className="c2"
  />
  <label
    className="c3"
    htmlFor="checkbox3"
  >
    Indeterminate
  </label>
</div>
`;

exports[`Checkbox should render checkbox 1`] = `
.c3 {
  cursor: pointer;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  padding-left: 8px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > input[type='checkbox']:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1 {
  position: absolute;
  z-index: 10;
  width: 16px;
  height: 16px;
  cursor: pointer;
  opacity: 0;
}

.c1:disabled ~ label {
  color: #e9eef2;
  cursor: not-allowed;
}

.c1:focus ~ span::before {
  box-shadow: 0 0 0 3px rgba(21,156,228,0.4);
}

.c1:checked ~ span::before {
  background-color: #159ce4;
  border-color: #159ce4;
}

.c1:disabled ~ span::before {
  border-color: #e9eef2;
}

.c1:checked:disabled ~ span::before {
  background-color: #e9eef2;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:hover:not(:disabled) ~ span::before {
  border-color: #159ce4;
}

.c2::before {
  display: inline-block;
  width: 16px;
  height: 16px;
  content: '';
  border: 2px solid #597183;
  border-radius: 2px;
  position: relative;
  top: 2px;
}

.c2::after {
  position: absolute;
  top: 7px;
  left: 3px;
  width: 10px;
  height: 6px;
  color: #ffffff;
  content: '';
  border-bottom: 2px solid;
  border-left: 2px solid;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

@media (prefers-reduced-motion:reduce) {

}

<div
  className="c0"
>
  <input
    className="c1"
    id="checkbox1"
    type="checkbox"
  />
  <span
    className="c2"
  />
  <label
    className="c3"
    htmlFor="checkbox1"
  >
    Checkbox
  </label>
</div>
`;
